rmf_traffic/include/rmf_traffic/Trajectory.hpp

```diff
@@ -43,270 +42,115 @@ class Trajectory
     /// are x and y velocities, while the third is rotational velocity about the
     /// z-axis.
-    Eigen::Vector3d get_finish_velocity() const;
+    Eigen::Vector3d velocity() const;
 
     /// Set the intended velocity of the robot at the end of this Trajectory
-    /// Segment.
+    /// Waypoint.
     ///
     /// This is a 2D homogeneous position. The first two values in the vector
     /// are x and y coordinates, while the third is rotation about the z-axis.
     ///
     /// \param[in] new_velocity
-    ///   The new finishing velocity for this Trajectory Segment.
-    Segment& set_finish_velocity(Eigen::Vector3d new_velocity);
+    ///   The new finishing velocity for this Trajectory Waypoint.
+    Waypoint& velocity(Eigen::Vector3d new_velocity);
 
-    /// Get the time that this Trajectory Segment is meant to finish.
-    Time get_finish_time() const;
+    /// Get the time that this Trajectory Waypoint is meant to finish.
+    Time time() const;
 
-    /// Change the finish time of this Trajectory Segment. Note that this
-    /// function will only affect this segment, and may cause this Segment to be
+    /// Change the finish time of this Trajectory Waypoint. Note that this
+    /// function will only affect this waypoint, and may cause this waypoint to be
     /// reordered within the Trajectory.
     ///
-    /// To change the finish time for this segment while preserving the relative
+    /// To change the finish time for this waypoint while preserving the relative
     /// times of all subsequent Trajectory segments, use adjust_finish_times()
     /// instead.
     ///
-    /// \warning If you change the finishing time value of this Segment such
-    /// that it falls directly on another Segment's finish time, you will get a
+    /// \warning If you change the finishing time value of this Waypoint such
+    /// that it falls directly on another Waypoint's finish time, you will get a
     /// std::invalid_argument exception, because discontinuous jumps are not
     /// supported, and indicate a significant mishandling of trajectory data,
     /// which is most likely a serious bug that should be remedied.
     ///
-    /// \note If this Segment's finish time crosses over another Segment's
+    /// \note If this Waypoint's finish time crosses over another Waypoint's
     /// finish time, that signficantly changes the topology of the Trajectory,
     /// because it will change the order in which the positions are passed
     /// through.
     ///
     /// \param[in] new_time
-    ///   The new finishing time for this Trajectory Segment.
+    ///   The new finishing time for this Trajectory Waypoint.
     ///
     /// \sa adjust_finish_times(Time new_time)
-    Segment& set_finish_time(Time new_time);
+    Waypoint& change_time(Time new_time);
 
-    /// Adjust the finishing time of this segment and all subsequent segments by
+    /// Adjust the finishing time of this waypoint and all subsequent segments by
```

Comment:
Same as above.
```suggestion
    /// Adjust the finishing time of this waypoint and all subsequent waypoints by
```

Reply:
I'm thinking we should also eliminate the use of "finishing" when referring to the time value of a waypoint. Now the trajectory elements simply represent waypoints instead of continuous segments, so there's no longer a point in qualifying the values with "finishing".

Reply:
Agreed!